src/course/application/course-enroll/course-enroll.ts

```diff
@@ -0,0 +1,36 @@
+import { CourseRepository } from '../../domain/repository/course-repository';
+import { EnrollmentRepository } from '../../domain/repository/enrollment-repository';
+import { Enrollment } from '../../domain/enrollment';
+import { CourseId } from '../../domain/course-id';
+import { UserId } from '../../../user/domain/user-id';
+import { CourseNotFoundError } from '../../domain/errors/course-not-found-error';
+import { AlreadyEnrolledError } from '../../domain/errors/already-enrolled-error';
+
+export class CourseEnroll {
+  constructor(
+    private courseRepository: CourseRepository,
+    private enrollmentRepository: EnrollmentRepository
+  ) {}
+
+  async execute(userId: string, courseId: string): Promise<Enrollment> {
+    const course = await this.courseRepository.getById(new CourseId(courseId));
+    if (!course) {
+      throw new CourseNotFoundError(`Course with id ${courseId} not found`);
+    }
+
+    const existingEnrollment =
+      await this.enrollmentRepository.findByUserAndCourse(
+        new UserId(userId),
+        new CourseId(courseId)
+      );
+
+    if (existingEnrollment) {
+      throw new AlreadyEnrolledError('User is already enrolled in this course');
+    }
+
+    const enrollmentId = `${userId}-${courseId}-${Date.now()}`;
```

Comment:
Enrollment id is generated as a concatenated string (`${userId}-${courseId}-${Date.now()}`) in the application layer. That approach is brittle and will not fit the Prisma schema which declares Enrollment.id as Int autoincrement.

**Sugerencia:**
```typescript
Align ID strategy across layers. If DB generates the id, do not create an id in the application layer; let the repository/persistence adapter populate it when persisting (or use UUIDs everywhere). If you prefer client-generated IDs, switch DB to String/UUID. Also consider potential predictability/collision implications and prefer UUID v4 when client-generated unique IDs are required.
```